app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulation.java

```diff
@@ -0,0 +1,173 @@
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMarshalling;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.Sets;
+
+/**
+ * A sub-population of the study participants who will receive a unique consent based on selection by a data group, 
+ * an application version, or both. 
+ */
+@DynamoDBTable(tableName = "Subpopulation")
+@BridgeTypeName("Subpopulation")
+public final class DynamoSubpopulation implements Subpopulation {
+
+    private String studyIdentifier;
+    private String guid;
+    private String name;
+    private String description;
+    private boolean required;
+    private boolean deleted;
+    private Integer minAppVersion;
+    private Integer maxAppVersion;
+    private Long version;
+    private Set<String> allOfGroups;
+    private Set<String> noneOfGroups;
+
+    public DynamoSubpopulation() {
+        this.allOfGroups = Sets.newHashSet();
+        this.noneOfGroups = Sets.newHashSet();
+    }
+    
+    @JsonIgnore
+    @Override
+    @DynamoDBHashKey
+    public String getStudyIdentifier() {
+        return studyIdentifier;
+    }
+    @Override
+    public void setStudyIdentifier(String studyIdentifier) {
+        this.studyIdentifier = studyIdentifier;
+    }
+    @Override
+    @DynamoDBRangeKey
+    public String getGuid() {
+        return guid;
+    }
+    @Override
+    public void setGuid(String guid) {
+        this.guid = guid;
+    }
+    @DynamoDBAttribute
+    @Override
+    public String getName() {
+        return name;
+    }
+    @Override
+    public void setName(String name) {
+        this.name = name;
+    }
+    @DynamoDBAttribute
+    @Override
+    public String getDescription() {
+        return description;
+    }
+    @Override
+    public void setDescription(String description) {
+        this.description = description;
+    }
+    @DynamoDBAttribute
+    @Override
+    public boolean isRequired() {
+        return required;
+    }
+    @Override
+    public void setRequired(boolean required) {
+        this.required = required;
+    }
+    @DynamoDBAttribute
+    @Override
+    public boolean isDeleted() {
+        return deleted;
+    }
+    @Override
+    public void setDeleted(boolean deleted) {
+        this.deleted = deleted;
+    }
+    @DynamoDBMarshalling(marshallerClass = StringSetMarshaller.class)
+    @DynamoDBAttribute
+    @Override
+    public Set<String> getAllOfGroups() {
+        return allOfGroups;
+    }
+    @Override
+    public void setAllOfGroups(Set<String> dataGroups) {
+        this.allOfGroups = (dataGroups == null) ? Sets.newHashSet() : Sets.newHashSet(dataGroups);
+    }
```

Comment:
If you feel strongly about this we can make them immutable. It means for this mutable object you have to do this:
subpop.setNoneOfGroups(Sets.newHashSet("group1"))
instead of
subpop.getNoneOfGroups().add("group1")
I find the latter easier to read, but can live with the former.